src/main/scala/Errors.scala

```diff
@@ -42,6 +42,9 @@ object Errors {
   case class AlreadyConsumed(id: Id, dim: Int, bank: Int) extends TypeError(
     s"Bank $bank in dimension $dim of $id already consumed.")
 
+  case class InvalidDynamicIndex(id:Id) extends TypeError(
+    s"Cannot dynamically access unbanked array $id.", id.pos)
```

Comment:
"Dynamic access of array $id requires unbanked array. Actual banking factor: $bf. Use a shrink view to create unbanked array"